node/node.go

```diff
@@ -18,26 +20,30 @@ import (
 	"github.com/synadia-io/nex/node/actors"
 )
 
+const ergoNodeName = "nex@localhost"
+
 type Node interface {
 	Validate() error
-	Start()
+	Start() error
 }
 
 type nexNode struct {
-	nc  *nats.Conn
 	ctx context.Context
+	nc  *nats.Conn
 
-	options *models.NodeOptions
+	options   *models.NodeOptions
+	publicKey nkeys.KeyPair
 }
 
-func NewNexNode(nc *nats.Conn, opts ...models.NodeOption) (Node, error) {
+func NewNexNode(publicKey nkeys.KeyPair, nc *nats.Conn, opts ...models.NodeOption) (Node, error) {
```

Comment:
Nit: if this is the whole pair then call it the server key instead of public key